Xamarin.PropertyEditing/ViewModels/RatioViewModel.cs

```diff
@@ -0,0 +1,169 @@
+			if (newDenominator < 1)
+				newDenominator = 1;
+
+			// Nothing selected, just workout out which value the caret is closest to, so we can increment it
+			if (selectionLength == 0) {
+				var sepLocation = GetSeparatorPosition (stringValue);
+				if (caretPosition <= sepLocation) {
+					Value = new CommonRatio (newNumerator, Value.Denominator, separator);
+				} else {
+					Value = new CommonRatio (Value.Numerator, newDenominator, separator);
+				}
+			} else {
+				// Increment both values.
+				Value = new CommonRatio (newNumerator, newDenominator, separator);
+			}
+		}
+
+		public string ValueString
+		{
+			get { return Value.StringValue; }
+			set {
+				if (Value.StringValue.Equals (value))
+					return;
+
+				SetRatioFromString (value);
+			}
+		}
+
+		public static char[] Separators = { ':', '/' };
+
+		public RatioViewModel (IPropertyInfo property, IEnumerable<IObjectEditor> editors)
+			: base (property, editors)
+		{
+		}
+
+		protected override void OnValueChanged ()
+		{
+			base.OnValueChanged ();
+			OnPropertyChanged (nameof (ValueString));
+		}
+
+		void SetRatioFromString (string value)
+		{
+			var strippedValue = value.Replace (" ", string.Empty);
+			bool parsed = false;
+			foreach (var separator in Separators) {
+				parsed = true;
+				var parts = strippedValue.Split (new[] { separator });
+				double numerator = 0f;
+				if (parts.Length == 2) { // We have a potential ratio, let's make sure
+					if (!IsDouble (parts[0], out numerator))
+						parsed = false;
+					double denominator = 0f;
+					if (!String.IsNullOrEmpty (parts[1]) && !IsDouble (parts[1], out denominator))
+						parsed = false;
+
+					if (parsed) {
+						Value = new CommonRatio (numerator, denominator, separator);
+						break;
+					}
+				} else if (parts.Length == 1) { // We have a potential whole number, let's make sure
+					if (!IsDouble (parts[0], out numerator)) {
+						parsed = false;
+					}
+					if (parsed) {
+						Value = new CommonRatio (numerator, 1, separator);
+						break;
+					}
+				}
+			}
+
+			if (!parsed) {
+				SetError (string.Format (LocalizationResources.InvalidRatio, value));
+			}
+		}
+
+		char GetSeparator (string value)
+		{
+			foreach (var separator in Separators) {
+				if (value.Contains (separator))
+					return separator;
+			}
+			return ':';
+		}
+
+		int GetSeparatorPosition (string finalString)
+		{
+			foreach (var separator in Separators) {
+				if (finalString.Contains (separator))
+					return finalString.IndexOf (separator);
+			}
+			return -1; // Not found
+		}
+
+		bool IsDouble (string stringValue, out double value)
+		{
+			//Checks parsing to number
+			if (!double.TryParse (stringValue, out value))
+				return false;
+					
+			//Checks if needs to be positive value
+			if (value < 0)
+				return false;
+					
+			//Checks a common validation
+			return CommonValidate (stringValue);
+		}
+
+		public static bool CommonValidate (string finalString)
```

Comment:
Maybe make this method name explicitly state usage. This is pretty vague.

Reply:
Agreed.